Include/httpClient/async.h

```diff
@@ -2,8 +2,6 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 #pragma once
 
-#include <httpClient/pal.h>
```

Comment:
why are we doing this?

Reply:
Why remove pal.h?  Because we want to be able to easily merge these files between libhttpclient and windows, and windows has no "libHttpClient/pal.h".  This header gets included from other uber headers from a user's perspective so it's not mandatory that it include pal.h.  That can come from higher level headers that include async.h.